agent/run.sh

```diff
@@ -376,6 +376,20 @@ setup_env()
         fi
     fi
 
+    if [[ -z ${CATTLE_AGENT_IP} ]]; then
+        info DETECTED CATTLE AGENT IP ${DETECTED_CATTLE_AGENT_IP}
+
+        local default_docker_addr="172.17.0.1"
```

Comment:
Not that it is changed often, but we might want to dynamically check for this. You can get the Docker bridge network device from `docker network inspect bridge -f '{{index .Options "com.docker.network.bridge.name"}}'` and then inspect the interface.

Reply:
@cloudnautique 
I also hope to do so. But this `run.sh` script runs in a container, which uses the docker `bridge` network, I cannot get `docker0` or  `com.docker.network.bridge.name` interface in this container.

I'm not sure if we can accept to use the `host` network in the container which runs `docker.register.command`.